### PR TITLE
feat: add DELETE /connectors/:id

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

add DELETE /connectors/:id API

## Testing

Passed ITs and UTs.

### CODE REVIEW

File: packages/core/src/routes/connector.ts [hunk 1]
```diff
@@ -208,4 +208,39 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+        });
+      }
+
+      const virtualConnectors = await loadConnectors();
+      const virtualConnector = virtualConnectors.find(
+        ({ metadata: { id } }) => id === databaseEntry.connectorId
+      );
+
+      if (virtualConnector) {
+        assertThat(!databaseEntry.enabled, 'connector.delete_enabled_connector_failed');
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:237]: How can the user set a connector's enabled to false?

**Review by wangsijie — COMMENTED** [review]
We have "enabled" and "disabled", now add "delete", it can be confusing, better confirm their usage.

File: packages/core/src/routes/connector.ts (hunk 1 above)
Comment (darcyYe) [reply packages/core/src/routes/connector.ts:237]: Users should go to the "Connector Details Page" to disable connectors and fix SIE before deleting a connector instance.
I think the guide should be given by AC after catching this error.

**Comment by darcyYe** [comment]
> We have "enabled" and "disabled", now add "delete", it can be confusing, better confirm their usage.

Per discussion, will remove `enabled` hence all checks on `enabled` should be removed as well.

File: packages/core/src/routes/connector.ts (hunk 1 above)
Comment (darcyYe) [reply packages/core/src/routes/connector.ts]: Removed this check.

File: packages/core/src/routes/connector.ts [hunk 2]
```diff
@@ -208,4 +208,31 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+  router.delete(
+    '/connectors/:id',
+    koaGuard({ params: object({ id: string().min(1) }) }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+
+      const databaseEntries = await findAllConnectors();
+      const databaseEntry = databaseEntries.find((entry) => entry.id === id);
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:221]: According to current design, all we need to do is call `deleteConnectorById`?

**Review by wangsijie — APPROVED** [review]

**Pull request merged**